packages/xo-web/src/xo-app/vm/tab-stats.js

```diff
@@ -14,169 +15,157 @@ import {
   XvdLineChart,
 } from 'xo-line-chart'
 
-export default injectIntl(
-  class VmStats extends Component {
-    constructor(props) {
-      super(props)
-      this.state.useCombinedValues = false
-    }
+export default class VmStats extends Component {
+  state = {
+    granularity: 'seconds',
+    useCombinedValues: false,
+  }
 
-    loop(vm = this.props.vm) {
-      if (this.cancel) {
-        this.cancel()
-      }
+  loop(vm = this.props.vm) {
+    if (this.cancel) {
+      this.cancel()
+    }
 
-      if (vm.power_state !== 'Running') {
-        return
-      }
+    if (vm.power_state !== 'Running') {
+      return
+    }
 
-      let cancelled = false
-      this.cancel = () => {
-        cancelled = true
-      }
+    let cancelled = false
+    this.cancel = () => {
+      cancelled = true
+    }
 
-      fetchVmStats(vm, this.state.granularity).then(stats => {
+    const { granularity } = this.state
+    const isLastDayGranularity = granularity === LAST_DAY
+    fetchVmStats(vm, isLastDayGranularity ? LAST_WEEK : granularity).then(
```

Comment:
What do you think of adding `fetchStats` in` common/xo`  which chooses the method to use depending on the type?

Reply:
What's the point of this method and how to use it?

Reply:
I see that the same treatment done after the resolution of different `fetch` that's why I said if you can do a single method that chooses the right method according to the type passed  and to not repeat the same treatment in many places.

Reply:
Do you mean something like `fetchStats(vm, granularity).then` or `fetchStats(vm, granularity)` will do all things?

Reply:
`fetchStats(object, granularity)`  do all things and object can be VM, host ... 
it can return stats or what you need.

Reply:
So the function `fetchStats` will take a `xoObject` and a `granularity` and returns `stats` or `contracted stats` in case of last day granularity ?

Reply:
yes. so what do you think?

Reply:
It's a good idea :+1:

IMHO, instead of passing an object with some properties, we can just pass an id and a type. What do you think?

Reply:
I think passing object it's better but do it as you see it. ^_^

Reply:
We need an id to fetch stats and a type to know which function to use. Can you please explain me why passing an object is better?

Reply:
Just to not destructure object evry time before using `fetchStats` and passing object to fucntion to do this.

Reply:
In our case we know the type we just need to get the id. 
IMHO, it's magic to pass an object to a public method with not used properties when we can just pass the needed params.
I can't say what it's better, i'll we let @pdonias give us its opinion and what we can do in this case.

Reply:
I'm not sure to understand the exact question. Can you reformulate?